File: 23-luhn-test/luhnTest.go
// Golang program to validate no. using luhn test
// Written by M.V.Harish Kumar on 20-05-2024
package main

import (
	"fmt"
)

func luhnTest(num int) bool {
	var num_list []int
	len := 0
	s1 := 0
	s2 := 0
	for num != 0 {
		num_list = append(num_list, num%10)
		num /= 10
		len++
	}
	for i := 1; i <= len; i++ {
		if i%2 != 0 {
			s1 += num_list[i-1]
		} else {
			tmp := num_list[i-1] * 2
			s2 += tmp/10 + tmp%10
		}
	}
	return (s1+s2)%10 == 0
}

func main() {
	var num int
	fmt.Print("Enter a number: ")
	fmt.Scanln(&num)
	fmt.Printf("Is the number %d validates luhn: %t\n", num, luhnTest(num))
}

File: 23-luhn-test/luhn_test.py
# Python program to validate no. with Luhn test
# Written by M.V.Harish Kumar on on 20-05-2024 

def luhn_test(number):
    num_list = [int(d) for d in number][::-1]
    s1 = sum(num_list[::2])
    s2 = sum([sum(divmod(n*2,10)) for n in num_list[1::2]])
    return (s1+s2) % 10 == 0

num = input("Enter a number: ")
print(f"Is the number {num} validates luhn: {luhn_test(num)}")
